packages/manager/apps/key-management-service/src/data/hooks/useDeleteOkmsService.ts

```diff
@@ -0,0 +1,48 @@
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import {
+  useMutation,
+  UseMutationOptions,
+  useQueryClient,
+} from '@tanstack/react-query';
+import { useContext } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import {
+  deleteService,
+  DeleteServiceResponse,
+  getOkmsServiceId,
+  getOkmsServiceIdQueryKey,
+} from '../api/okmsService';
+
+export type DeleteOkmsServiceMutationParams = {
+  resourceName: string;
+};
+
+export const deleteOkmsServiceMutationKey = ['delete-okms-service'];
+
+export const useDeleteOkmsService = (
+  options: Partial<
+    UseMutationOptions<
+      ApiResponse<DeleteServiceResponse>,
+      Error,
+      DeleteOkmsServiceMutationParams
+    >
+  >,
+) => {
+  const queryClient = useQueryClient();
+  const { ovhSubsidiary } = useContext(ShellContext).environment.getUser();
+
+  return useMutation({
+    mutationKey: deleteOkmsServiceMutationKey,
+    mutationFn: async ({ resourceName }: DeleteOkmsServiceMutationParams) => {
+      const { data } = await queryClient.fetchQuery<
+        ApiResponse<number[]>,
+        ApiError
+      >({
+        queryKey: getOkmsServiceIdQueryKey(resourceName),
+        queryFn: () => getOkmsServiceId(resourceName),
+      });
+      return deleteService(data[0], ovhSubsidiary);
+    },
```

Comment:
I don't really see why we cache the serviceId here, it adds complexity and the cached value is not used anywhere.